helix-term/src/commands.rs

```diff
@@ -1323,14 +1319,35 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
             .collect()
     };
 
+    let doc_path = doc
+        .relative_path()
+        .map(|path| path.parent().unwrap().to_path_buf())
+        .unwrap_or_default();
+
+    let mut search_paths = vec![doc_path];
+
+    let cwd = helix_stdx::env::current_working_dir();
+    if cwd.exists() {
+        search_paths.push(cwd);
+    }
+    // FIXME: Also include project root? Or just use project root, not cwd?
```

Comment:
I could not quickly figure out how to find the LSP root for the document, but if available, this may be a better (or at least a valid additional) choice.